### ISSUE TITLE
Cannot resolve google.cloud imports

### ISSUE THREAD

**Issue by JherezTaylor** [microsoft/pyright#2563]
**Describe the bug**
The latest release 1.1.187 appears to introduce a regression where google.cloud imports are not resolved.
This appears to be similar to whats described in https://github.com/microsoft/pyright/issues/2113.

**To Reproduce**
- npm install pyright 1.1.187
- have a file that imports the google cloud api `from google.cloud import storage`
- run pyright
- Error: `"storage" is unknown import symbol (reportGeneralTypeIssues)`

**Expected behavior**
No errors. Reverting to 1.1.186 resolves the issue

**Screenshots or Code**
```python
from google.cloud import storage
```

Edited for brevity
```bash
 Attempting to resolve stub package using root path '...../.pyenv/versions/3.9.5/envs/.../lib/python3.9/site-packages'
  Attempting to resolve using root path '..../.pyenv/versions/3.9.5/envs/.../lib/python3.9/site-packages'
  Looking for typeshed path
  Looking for typeshed stubs path
  Typeshed path not found
  ........:11:26 - error: "storage" is unknown import symbol (reportGeneralTypeIssues)
1 errors, 0 warnings, 0 infos
```

**VS Code extension or command-line**
cli 1.1.186 via npm

**Additional context**
I understand there's some work happening in https://github.com/python/typeshed/pull/6106 that's related to this, but should the guidance be to pin the pyright version until this is resolved? Open to suggestions.

**Comment by erictraut** [comment]
This surfaced because of a recent change to typeshed stubs. I think there's a bug in the new stubs (they're supposed to be a namespace stub package, but they contain an `__init__.pyi` file). There was also a bug in pyright's import resolution logic.

@heejaechang, please review [this commit](https://github.com/microsoft/pyright/commit/f10e8f8326aa329f20350bca16b2d2743f5b51f2) and let me know if you have any concerns.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.188, which I just published. It will also be included in the next release of pylance.

**Comment by John15321** [comment]
@erictraut this is still an issue, and looks like its everywhere. No matter if I use PyCharm vscode or whatever

**Comment by erictraut** [comment]
@John15321, this bug was fixed a while ago. If you are experiencing a similar problem, please open a separate bug. If you're seeing the same behavior across multiple type checkers (e.g. pyright, PyCharm, and mypy), then it's unlikely a bug in all of these type checkers but rather a problem with the stubs or your project.
